### PR TITLE
feat: add endpoint to fetch list contributors highlight tagged repos

### LINKED ISSUE #432: Feature: add endpoint to fetch tagged repos of list contributors highlights
### Type of feature

🍕 Feature

### Current behavior

We currently have something similar with the general highlights service.

This is needed to filter highlights by tagged repos in https://github.com/open-sauced/app/issues/2069

### Suggested solution

create new endpoint to fetch tagged repos for lists contributors highlights

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR adds a new endpoint `v1/lists/{id}/contributors/highlights/tagged-repos` to fetch lists contributors highlights tagged repos

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes #432 

## Mobile & Desktop Screenshots/Recordings

image

### Example response
image

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by jpmcb — COMMENTED** [review]
Question: I'm confused what the benefit of having this as a separate endpoint is? If we are returning the `tagged_repos` and using that on the frontend to filter a query to `/:id/contributors/highlights`, why not add a new option to the `HighlightOptionsDto` to only return highlights where where they match the given tagged_repo?

I'm probably not understanding the use case here but would like to know if there's an easy optimization win here!

**Review by brandonroberts — APPROVED** [review]

File: src/user-lists/user-list.service.ts [hunk 1]
```diff
@@ -277,6 +278,35 @@ export class UserListService {
...
+  async findListContributorsHighlightedRepos(
+    pageOptionsDto: PageOptionsDto,
+    listId: string
+  ): Promise<PageDto<DbUserHighlightRepo>> {
+    const startDate = GetPrevDateISOString(pageOptionsDto.prev_days_start_date);
+    const range = pageOptionsDto.range ?? 30;
+    const orderBy = pageOptionsDto.orderDirection ?? "DESC";
+    const queryBuilder = this.userHighlightRepository.createQueryBuilder("user_highlights");
+
+    queryBuilder.select("DISTINCT UNNEST(user_highlights.tagged_repos) AS full_name");
```
[the remark this answers is not in the document]
Comment (OgDev-01) [reply src/user-lists/user-list.service.ts:290]: The unnest() function in PostgreSQL is used to expand an array into a set of rows. It takes an array as input and returns a new table where each element of the array occupies a separate row. This function is particularly useful when dealing with arrays in PostgreSQL.

Read about it here after researching
https://www.w3resource.com/PostgreSQL/postgresql_unnest-function.php

**Comment by OgDev-01** [comment]
> Question: I'm confused what the benefit of having this as a separate endpoint is? If we are returning the `tagged_repos` and using that on the frontend to filter a query to `/:id/contributors/highlights`, why not add a new option to the `HighlightOptionsDto` to only return highlights where where they match the given tagged_repo?
> 
> I'm probably not understanding the use case here but would like to know if there's an easy optimization win here!

Good question 👍

Here is a breakdown if why we need a separate endpoint for this.

1. From the frontend, we have a select component that needs an array of options(repos) for users to select and filter the highlights using the selected repo. Hard coding this won't scale

2. Having this endpoint will help prevent users from selecting repos that does not exist in the highlights since this will keep updating based on highlights for the contributors in the list.

To summarize,  we want to make the select options accurate and prevent users from selecting a filter that has no result for tagged repos.

**Review by jpmcb — APPROVED** [review]
👏🏼

**Review by nickytonline — APPROVED** [review]
🚀
